From c82b7dad71c6917c1ffa387b87de50a7d786e403 Mon Sep 17 00:00:00 2001
Message-Id: <c82b7dad71c6917c1ffa387b87de50a7d786e403.1482436097.git.ymankad@redhat.com>
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Fri, 16 Dec 2016 11:56:24 -0500
Subject: [CHANGE 1/4] rename qemu_aio_context() to match upstream
To: ymankad@redhat.com

RH-Author: Paolo Bonzini <pbonzini@redhat.com>
Message-id: <20161216115627.27874-2-pbonzini@redhat.com>
Patchwork-id: 73073
O-Subject: [RHEL6.9 qemu-kvm PATCH 1/4] rename qemu_aio_context() to match upstream
Bugzilla: 876993
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: John Snow <jsnow@redhat.com>
RH-Acked-by: Fam Zheng <famz@redhat.com>

Upstream, the name of the function is qemu_get_aio_context().  Since we
have to make it non-static, update to match.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Yash Mankad <ymankad@redhat.com>
---
 async.c    | 14 +++++++-------
 qemu-aio.h |  1 +
 2 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/async.c b/async.c
index 69539ba..a43f359 100644
--- a/async.c
+++ b/async.c
@@ -226,7 +226,7 @@ void aio_flush(AioContext *ctx)
  */
 static AioContext *__qemu_aio_context;
 
-static AioContext *qemu_aio_context(void)
+AioContext *qemu_get_aio_context(void)
 {
     if (!__qemu_aio_context)
         __qemu_aio_context = aio_context_new();
@@ -236,28 +236,28 @@ static AioContext *qemu_aio_context(void)
 
 QEMUBH *qemu_bh_new(QEMUBHFunc *cb, void *opaque)
 {
-    return aio_bh_new(qemu_aio_context(), cb, opaque);
+    return aio_bh_new(qemu_get_aio_context(), cb, opaque);
 }
 
 int qemu_bh_poll(void)
 {
-     return aio_bh_poll(qemu_aio_context());
+     return aio_bh_poll(qemu_get_aio_context());
 }
 
 void qemu_bh_update_timeout(uint32_t *timeout)
 {
-    aio_bh_update_timeout(qemu_aio_context(), timeout);
+    aio_bh_update_timeout(qemu_get_aio_context(), timeout);
 }
 
 
 void qemu_aio_flush(void)
 {
-    aio_flush(qemu_aio_context());
+    aio_flush(qemu_get_aio_context());
 }
 
 bool qemu_aio_wait(void)
 {
-    return aio_poll(qemu_aio_context(), true);
+    return aio_poll(qemu_get_aio_context(), true);
 }
 
 #ifdef CONFIG_POSIX
@@ -267,7 +267,7 @@ void qemu_aio_set_fd_handler(int fd,
                              AioFlushHandler *io_flush,
                              void *opaque)
 {
-    aio_set_fd_handler(qemu_aio_context(), fd, io_read, io_write, io_flush,
+    aio_set_fd_handler(qemu_get_aio_context(), fd, io_read, io_write, io_flush,
                        opaque);
 
     qemu_set_fd_handler2(fd, NULL, io_read, io_write, opaque);
diff --git a/qemu-aio.h b/qemu-aio.h
index 4297fe6..4e660eb 100644
--- a/qemu-aio.h
+++ b/qemu-aio.h
@@ -203,6 +203,7 @@ GSource *aio_get_g_source(AioContext *ctx);
 
 void qemu_aio_flush(void);
 bool qemu_aio_wait(void);
+AioContext *qemu_get_aio_context(void);
 
 void qemu_aio_set_fd_handler(int fd,
                              IOHandler *io_read,
-- 
2.7.4

